### ISSUE TITLE
ShadowNotifiction#getContentText() returns SubText if notification object has one.

### ISSUE THREAD

**Issue by martinthuresson** [robolectric/robolectric#2470]
### Description

The implementation of ShadowNotifiction#getContentText relies on the TextView having the id "id.text". I found that was not true if Sub Text is set on the notification. In this case, the Sub Text gets "id.text" and the Content Text gets the id "id.text2".
### Steps to Reproduce

After a test failing after the update  https://github.com/robolectric/robolectric/commit/9d68c9a82b0c22bb716270222ec8174a1da48eaf
 I used hierarchyviewer to verify the id issue on a device running Lollipop.

**Comment by jy-z-cheng** [comment]
The implementations in ```android.app.Notification``` confirms with @martinthuresson's finding, see:

``` java
if (mContentText != null) {
  contentView.setTextViewText(R.id.text, processLegacyText(mContentText));
  showLine3 = true;
}

...

// Need to show three lines?
if (mSubText != null) {
    contentView.setTextViewText(R.id.text, processLegacyText(mSubText));
    if (mContentText != null) {
        contentView.setTextViewText(R.id.text2, processLegacyText(mContentText));
        contentView.setViewVisibility(R.id.text2, View.VISIBLE);
        showLine2 = true;
        contentTextInLine2 = true;
    } else {
        contentView.setViewVisibility(R.id.text2, View.GONE);
    }
} else {
    contentView.setViewVisibility(R.id.text2, View.GONE);
    if (hasProgress && (mProgressMax != 0 || mProgressIndeterminate)) {
        contentView.setViewVisibility(R.id.progress, View.VISIBLE);
        contentView.setProgressBar(
                R.id.progress, mProgressMax, mProgress, mProgressIndeterminate);
        contentView.setProgressBackgroundTintList(
                R.id.progress, ColorStateList.valueOf(mContext.getColor(
                        R.color.notification_progress_background_color)));
        if (mColor != COLOR_DEFAULT) {
            ColorStateList colorStateList = ColorStateList.valueOf(mColor);
            contentView.setProgressTintList(R.id.progress, colorStateList);
            contentView.setProgressIndeterminateTintList(R.id.progress, colorStateList);
        }
        showLine2 = true;
    } else {
        contentView.setViewVisibility(R.id.progress, View.GONE);
    }
}
```


There are three scenarios:

1. If only **contentText** is set, the content text is set to the texview with "R.id.text"
2. If both **contentText** and **subText** are set, the sub text is set to the texview with "R.id.text" and  the content text is set to the texview with "R.id.text2"
3. If only **subText** is set, the sub text is set to the texview with "R.id.text"

It does not seem like there is a way to distinguish 1 and 3 without shadowing the Notification.Builder.

**Comment by xian** [comment]
@jongerrish is this addressed by the changes to `ShadowNotification`?

**Comment by jongerrish** [comment]
Not 100% sure, I think it might work on newer versions of the SDK, N+ specifically, but older versions still stash the values into content views rather than a bundle.

**Comment by pivotal-james-zcheng** [comment]
@jongerrish @xian The changes to `ShadowNotification` does not address this issue. The current implementation does not have good support for notifications with sub texts.

**Comment by pivotal-james-zcheng** [comment]
The three scenarios described previously only applies to pre-N versions. In pre-N versions, because the values are stashed into content views, it is difficult to distinguish contentText from content subText. As a result, in my original PR, I proposed to rename `getContentText` to `getCombinedContentText` to have it return both contentText and subText. 

In N+, I noticed `realNotification.extras.getString(Notification.EXTRA_SUB_TEXT)` would actually directly give us the subText. As such, we can directly add a `getSubText` to ShadowNotification in N+. @jongerrish @xian Is it possible to add a method to a Shadow for specific APIs? Is that a recommended approach?

Currently, to prevent introducing API breaking changes, I did not rename `getContentText` in my PR. However, that name is inaccurate as it returns both contentText and subText. Suggestions and comments are welcomed!
